### ISSUE TITLE
As an admin I'd like to test easy admin access to appropriate roles/privileges in Open Forest so that future business can be performed efficiently.

### ISSUE THREAD

**Issue by ASprinkle** [USDAForestService/fs-open-forest-platform#1182]
## Notes
Connected to program board https://code.fs.usda.gov/forest-service/Open-Forest-Program/issues/190#issue-20798

## Acceptance Criteria
- [ ] We have a way of testing EEMS integration in the staging environment

## Tasks
- [ ] Develop front end to test EEMS integration
- [ ] PO approved

## Definition of Done
- [ ] Pull requests meet technical definition of done
- [ ] Compare finished design with mockup
- [ ] Usability tested

**Comment by abdul-fs** [comment]
Update: we created a branch Eauth_add_Roles and thanks to sirisha for adding code header in the code. The header will give us who as admin logged into openforest with their role. there still more work to be done at developer side.  We are currently testing this in dev environment.

**Comment by mwreiss** [comment]
Thank you @abdul-fs and @sirishafs!  When the time comes @JonathanLerner54  and I are ready to put a test user in a role within the EEMS management system.  So far we are set in the EEMS CERT environment.

**Comment by abdul-fs** [comment]
Upate: usdaapproles" are empty and working with ICAM on getting values for this attribute

**Comment by ASprinkle** [comment]
Update from Stand up: Waiting for eAuth, setting up meeting with EEMS to add roles. Matt and Jon are not able to see users portion in EEMS, and they are working to resolve that access/roles issue on our end.

**Comment by ASprinkle** [comment]
Update from stand up: Super User Roles added for Sirisha. Looking at code changes to get rid of whitelist, requires adding new attributes to the ICAM, meeting with ICAM team tomorrow for adding forest information (a required piece).

**Comment by ASprinkle** [comment]
update from stand up: deploying changes in dev; once done we can test

**Comment by abdul-fs** [comment]
Update: we made progress where if the user is not assigned Super user in eauth(EEMS) then the role would be user otherwise an admin.

**Comment by abdul-fs** [comment]
Update: In order to get rid of whitelist completely we need to have the forest info added in eems

**Comment by ASprinkle** [comment]
@abdul-fs  ok. How do we make sure we get the forest info added in EEMS? Is there something that I should be doing, or someone else on the team?

**Comment by abdul-fs** [comment]
@ASprinkle @mwreiss  Jon and team are working with EEMs. We sent them dev whitelist on last meeting.

**Comment by ASprinkle** [comment]
Update from Stand Up: Sirisha has base code in place. Outstanding piece is all the additional roles that need to be created (Since it's all in a flat structure now). Matt et al is talking to ICAM team about adding additional attributes for roles.  They have copy of white list so that they can copy the information over.  Code on Dev reads roles from eAuth header, and will decide if user is Admin or not. Only dependency right now is that user is on white list. Tests are working. 
 Once ICAM folks add info on header, we can move off of white list completely. Continuing to make progress.  There may be approximately 300+ roles looking at all forests... have 20-30 for pilot forests now.

**Comment by ASprinkle** [comment]
From stand up: Waiting on EEMS to get back to us.  EEMS going through some staffing transition right now - @smahmudFS is helping expedite coordination, haven't received any replies from Matt's requests for updates/info.

**Comment by mwreiss** [comment]
I'm seeing all forest show up after adding super user role.  Phase I progress seems to be working as expected.  Phase II is required to lock a user down to a specific forest POC1 or POC2 roles.

![image](https://user-images.githubusercontent.com/50459930/78151856-a710b980-73fe-11ea-9c9e-d2a74aafb690.png)

**Comment by mwreiss** [comment]
Phase II testing:

As a POC1 role I'm able to perform all admin functions. I should only be able to generate reports and view feedback.
  
As a POC2 role I'm able to change season dates, that should be removed.

![image](https://user-images.githubusercontent.com/50459930/78795874-26f8cf80-797b-11ea-9881-ed3eafd34727.png)

**Comment by briandavidson** [comment]
Were there changes to the POC1 role? You're saying now that POC1 should only be able to generate reports and view feedback, but that is not consistent with the User Entity Diagram I received:

[cid:22308e44-fd20-4eef-ae6c-6ac149f7ead2]
________________________________

**Comment by mwreiss** [comment]
Brian, please look and see that the comment was updated.

**Comment by carlsonem** [comment]
@briandavidson can you please push your latest code on this issue to staging? There is a bug in dev, but we think it might be a result of not all of the connections are setup in the dev environment, so it might simply be resolved when deployed to staging.

**Comment by carlsonem** [comment]
Need to add the regional roles in place

**Comment by briandavidson** [comment]
> Need to add the regional roles in place

@carlsonem at one point we decided to break out regional roles into it's own [issue](https://github.com/USDAForestService/fs-open-forest-platform/issues/1300)

This issue is complete and covers getting us to parity with whitelist functionality, as well as the additional functionality for handling POC roles.

**Comment by briandavidson** [comment]
@abdul-fs @kilara77 @mwreiss @carlsonem @smahmudFS 

The code for this issue has been merged into the `dev` branch and we can proceed with removal of the whitelist on your folks end. 🎉

**Comment by abdul-fs** [comment]
@briandavidson Thanks brian. the whitelist has been removed from dev and staging. @mwreiss we need to test this with internal and regional admin to confirm the changes and proceed for production.

**Comment by mwreiss** [comment]
I'm set as FS_Open-Forest_R06 role.
![image](https://user-images.githubusercontent.com/50459930/82327969-d5455b00-99a4-11ea-93f5-8cacc0718d5e.png)

I see forests for Region 6
![image](https://user-images.githubusercontent.com/50459930/82328064-0160dc00-99a5-11ea-9913-38ddf0ffa3b3.png)

![image](https://user-images.githubusercontent.com/50459930/82328133-18073300-99a5-11ea-9cf2-8d4876593784.png)

**Comment by mwreiss** [comment]
I'm set as FS_Open-Forest_R01 role:

![image](https://user-images.githubusercontent.com/50459930/82328956-30c41880-99a6-11ea-9503-319eb115e1eb.png)


I see all forests.  I expect to see just region 1 forest.  Flathead

![image](https://user-images.githubusercontent.com/50459930/82328789-e9d62300-99a5-11ea-8ef5-5b80adabc62e.png)

**Comment by mwreiss** [comment]
@abdul-fs @briandavidson did the recent changes include the regional roles and removal of whitelist.  Or just the removal of the whitelist?  Thank you!

**Comment by briandavidson** [comment]
@mwreiss regional roles are outside the scope of this issue. You're looking for [1300](https://github.com/USDAForestService/fs-open-forest-platform/issues/1300).

**Comment by abdul-fs** [comment]
@Dmac26 @mwreiss Poc1, poc2 and superuser work as expected. we are good to user management with internal forest users. they access is granted through icams identity management. schedule test with user are tested already but it is good to check with all internal users with admin rights.

**Comment by Dmac26** [comment]
Hey @aaronburk  -  it sounds like we haven't done end-user testing on this card and will need to complete that to fully validate. Is that something I can set up? And if so, I'm thinking Hernandez, Rebekah - FS would be a good one to reach out to seeing as she has set up these testing sessions before. Thank you for any input!
